### PR TITLE
[ICD] Finish ICDM server refactor

### LINKED ISSUE #30631: [ICD] Refactor ICDManagementServer functions to have less inputs
#### Description
See https://github.com/project-chip/connectedhomeip/pull/30618#discussion_r1403689617

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
Finishes the ICDM server refactor that was started in the #30618 

* Move commandData processing inside the associated function
* Move Admin check inside the the associated function
* rename variables to match SDK casing

fixes #30631 

#### Tests
PR doesn't do any functionnal changes.
Using CI to validate no regressions where caused.

### CODE REVIEW

File: src/app/clusters/icd-management-server/icd-management-server.cpp [hunk 1]
```diff
@@ -304,18 +322,19 @@ void ICDManagementServer::TriggerICDMTableUpdatedEvent()
     ICDNotifier::GetInstance().BroadcastICDManagementEvent(ICDListener::ICDManagementEvents::kTableUpdated);
 }
 
-CHIP_ERROR ICDManagementServer::CheckAdmin(CommandHandler * commandObj, const ConcreteCommandPath & commandPath, bool & isAdmin)
+CHIP_ERROR ICDManagementServer::CheckAdmin(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
```
Comment (andy31415) [inline src/app/clusters/icd-management-server/icd-management-server.cpp:325]: does this method use `this` in any way? If not could we remove it from the class and place it in an anonymous namespace?

**Review by andy31415 — APPROVED** [review]

File: src/app/clusters/icd-management-server/icd-management-server.cpp (hunk 1 above)
Comment (mkardous-silabs) [reply src/app/clusters/icd-management-server/icd-management-server.cpp:325]: No, function doesn't need `this`. We could move it to an anonymous namespace or make it static within the class. Don't have a preference.

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/clusters/icd-management-server/icd-management-server.cpp (hunk 1 above)
Comment (andy31415) [reply src/app/clusters/icd-management-server/icd-management-server.cpp:325]: I think it has no reason to appear in the header then, so we may as well use anon namespace for that reason (smaller header is smaller api for anyone reading the code)

Comment (mkardous-silabs) [reply src/app/clusters/icd-management-server/icd-management-server.cpp]: done

File: src/app/clusters/icd-management-server/icd-management-server.cpp [hunk 2]
```diff
@@ -188,6 +188,30 @@ class IcdManagementFabricDelegate : public FabricTable::Delegate
 IcdManagementFabricDelegate gFabricDelegate;
 IcdManagementAttributeAccess gAttribute;
 
+/**
+ * @brief Function checks if the client as admin permissions to the cluster in the commandPath
```
Comment (bzbarsky-apple) [inline src/app/clusters/icd-management-server/icd-management-server.cpp:192]: "has admin", no?

Comment (mkardous-silabs) [reply src/app/clusters/icd-management-server/icd-management-server.cpp:192]: Yes, I'll open the pr to fix it.

Comment (mkardous-silabs) [reply src/app/clusters/icd-management-server/icd-management-server.cpp:192]: pr is open #30848

**Pull request merged**